### PR TITLE
feat(query): impl LocalShuffle

### PR DESCRIPTION
**Pull request by Freejww**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary
Add localshuffle to parallel sort in window.

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16055)

### CODE REVIEW

File: tests/sqllogictests/suites/query/window_function/window_ignore_nulls.test [hunk 1]
```diff
@@ -20,7 +20,7 @@ SELECT
     ORDER BY id
     ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING
   ) AS last_order_id
-FROM issue2549
+FROM issue2549 order by 1,2,3,4
```
Comment (TCeason) [inline tests/sqllogictests/suites/query/window_function/window_ignore_nulls.test:23]: You mean order by id?

Comment (Freejww) [reply tests/sqllogictests/suites/query/window_function/window_ignore_nulls.test:23]: yes, to keep result consistent.

File: src/query/pipeline/transforms/src/processors/transforms/transform_local_shuffle.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            },
+        )))
+    }
+}
+
+impl Transform for TransformLocalShuffle {
+    const NAME: &'static str = "LocalShuffleTransform";
+
+    fn transform(&mut self, block: DataBlock) -> Result<DataBlock> {
+        let block = block.convert_to_full();
```
Comment (sundy-li) [inline src/query/pipeline/transforms/src/processors/transforms/transform_local_shuffle.rs:59]: convert_to_full only need for `group_hash_columns`, the other column don't need to convert to full column.

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
This PR passed the wizard select tests:
https://github.com/datafuselabs/wizard/blob/main/checksb/sql/selects/check.sql

**Pull request merged**